zaza/openstack/charm_tests/ceph/rbd_mirror/tests.py

```diff
@@ -153,6 +154,12 @@ def create_volume(cinder, volume_params, retry=20):
     return create_volume(cinder, volume_params)
 
 
+def remove_internal_pools(pools):
+    """Exclude the internal pools from the passed dict."""
+    for pool in INTERNAL_POOLS:
+        pools.pop(pool, None)
```

Comment:
May I caution about writing this kind of function.  Essentially, it's using the `pools` parameter as both and in and out parameter, which is generally considered bad practice as it breaks the principle of least surprise.  It also increases cognitive load, particularly as the function's parameters are not described.  It would be much better to implement the function the following way which makes it pure (side-effect free) and thus easier to reason about.

```python
def remove_internal_pools(pools):
    """Remove the internal pools from the passed dictionary.

    :param pools: the pools passed.
    :type pools: Dict[str, Any]
    :returns: pools with the INTERNAL pools remove, if they were present
    :rtype: Dict[str, Any]
    """
    _pools = pools.copy()  # top level copy only
    for pool in INTERNAL_POOLS:
        _pools.pop(pool, None)
    return _pools

# ... later

site_a_pools = remove_internal_pools(site_a_pools)
site_b_pools = remove_internal_pools(site_b_pools)
```

Now it's really obvious that `site_a_pools` and `site_b_pools` are being updated purposefully.